sycl/include/sycl/ext/intel/esimd/math.hpp

```diff
@@ -342,26 +342,39 @@ std::enable_if_t<detail::is_esimd_scalar<T>::value, T>(min)(T src0, T src1,
+  __ESIMD_DNS::vector_type_t<__ESIMD_DNS::__raw_t<T>, N> res =                 \
+      __spirv_ocl_native_##iname<__ESIMD_DNS::__raw_t<T>, N>(src.data());      \
+  if constexpr (std::is_same_v<Sat, saturation_off_tag>)                       \
+    return res;                                                                \
+  else                                                                         \
+    return esimd::saturate<T>(simd<T, N>(res));
+#define __ESIMD_SC_IMPL(T, name, iname)                                        \
+  __ESIMD_DNS::__raw_t<T> res =                                                \
+      __spirv_ocl_native_##iname<__ESIMD_DNS::__raw_t<T>>(src);                \
+  if constexpr (std::is_same_v<Sat, saturation_off_tag>)                       \
+    return res;                                                                \
+  else                                                                         \
+    return esimd::saturate<T>(simd<T, 1>(res))[0];
+#else
+#define __ESIMD_VEC_IMPL(T, name, iname) return 0;
+#define __ESIMD_SC_IMPL(T, name, iname) return 0;
+#endif // __SYCL_DEVICE_ONLY__
+
 #define __ESIMD_UNARY_INTRINSIC_DEF(COND, name, iname)                         \
   /** Vector version.                                                       */ \
   template <class T, int N, class Sat = saturation_off_tag,                    \
             class = std::enable_if_t<COND>>                                    \
   __ESIMD_API simd<T, N> name(simd<T, N> src, Sat sat = {}) {                  \
-    __ESIMD_DNS::vector_type_t<__ESIMD_DNS::__raw_t<T>, N> res =               \
-        __esimd_##iname<T, N>(src.data());                                     \
-    if constexpr (std::is_same_v<Sat, saturation_off_tag>)                     \
-      return res;                                                              \
-    else                                                                       \
-      return esimd::saturate<T>(simd<T, N>(res));                              \
+    __ESIMD_VEC_IMPL(T, name, iname)                                           \
   }                                                                            \
                                                                                \
   /** Scalar version.                                                       */ \
   template <typename T, class Sat = saturation_off_tag,                        \
             class = std::enable_if_t<COND>>                                    \
   __ESIMD_API T name(T src, Sat sat = {}) {                                    \
-    simd<T, 1> src_vec = src;                                                  \
-    simd<T, 1> res = name<T, 1>(src_vec, sat);                                 \
-    return res[0];                                                             \
+    __ESIMD_SC_IMPL(T, name, iname)                                            \
```

Comment:
minor comment. These macros are never used in any long lines. If so, why not call them: `__ESIMD_VECTOR_IMPL` and `__ESIMD_SCALAR_IMPL`?